pkg/server/api/api.go

```diff
@@ -26,6 +30,33 @@ func ProtoFromID(id spiffeid.ID) *types.SPIFFEID {
 	}
 }
 
+// AuthorizedEntryFetcher is the interface to fetch authorized entries
+type AuthorizedEntryFetcher interface {
+	// FetchAuthorizedEntries fetches the entries that the caller
+	// that is extracted from the context is authorized for
+	FetchAuthorizedEntries(ctx context.Context) ([]*types.Entry, error)
+}
+
+// FetchAuthEntries fetches authorized entries using the provided
+// implementation of the AuthorizedEntryFetcher interface.
+// The provided implementation of that interface must ensure that
+// the authorized entries are fetched from the caller that
+// is obtained from the context.
+func FetchAuthEntries(ctx context.Context, log logrus.FieldLogger, ef AuthorizedEntryFetcher) (map[string]*types.Entry, error) {
```

Comment:
I understand wanting to DRY this up, but I think we should wait. Both places that use this (so far) have different needs. One wants to build a map, the other doesn't. And outside of building the map, there isn't much to share here (since the AuthorizedEntryFetcher does all the work).

Reply:
I was debating myself whether this should be here or in the specific packages. It's clear for me now that two different implementations are needed.